### PR TITLE
Hkh/nosql

### PR DESCRIPTION
**Pull request by HosseinHeris**
# Description
Please include a summary of the change or issue fix.

## Type of change 
NOSQL feature is added: preprocessing + main flow files + vectordb samples + experimentation tracking + tests + workflows + readme + environment yaml + requirement.txt (in main flow folder) 
Please remove options that are not relevant. 
- [x] New feature: not breaking, new functionality 
- [ ] New feature: may cause existing code to not work as expected 
- [ ] Bug fix 
- [ ] Needs documentation update 

## Test and checks 
Please describe how you've tested your file 
- [x] tested via a notebook
- [x] tested via a testflow 
- [x] only made sure the code is written correctly 
Checks
- [x] commented code for complex areas
- [x] updated readme with the new changes 
- [x] changes not generating errors with the main flows in rag-vectorsearchtype directories (e.g rag-cosmos-mongo)
- [x] I have tested my implementation 
- [x] I have updated environment.yaml file with any additional dependencies
- [X] I have updated the requirement.txt file for any additional dependencies.

### CODE REVIEW

File: financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb
Comment (journeyman-msft) [inline financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb:1]: There is a comment related to pgvector in the function called `retrieve_data`.

 # Register 'pgvector' type for the 'embedding' column

Comment (journeyman-msft) [inline financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb:1]: Should document for users that you need to enabled Vector Search Preview feature: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search

**Review by journeyman-msft — CHANGES_REQUESTED** [review]
see comments, but overall looking good.

File: financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb
Comment (HosseinHeris) [reply financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb:1]: done!

Comment (HosseinHeris) [reply financial_transcripts/vectordb-tools/cosmos-nosql/preprocessing/preprocessing_cosmos_nosql.ipynb:1]: done!

**Review by journeyman-msft — APPROVED** [review]

**Pull request merged**